PocketKit/Sources/PocketKit/Collections/CollectionViewModel.swift

```diff
@@ -89,14 +98,17 @@ class CollectionViewModel {
     }
 
     func fetch() {
+        guard !isOffline else {
+            self.snapshot = errorSnapshot()
+            return
+        }
         // TODO: NATIVE COLLECTIONS - Update to fetch locally first
         Task {
             do {
                 self.collection = try await source.fetchCollection(by: slug)
-                let snapshot = buildSnapshot()
-                guard snapshot.numberOfItems != 0 else { return }
-                self.snapshot = snapshot
+                updateSnapshot()
             } catch {
+                self.snapshot = errorSnapshot()
                 Log.capture(message: "Failed to fetch details for CollectionViewModel: \(error)")
```

Comment:
This code will change with the use of Core Data and an `NSFetchedResultController` containing the stories